File: Trunk/Modules/HTML/Components/DataProviderPro.vb
'
' DotNetNuke® - http://www.dotnetnuke.com
' Copyright (c) 2002-2009 
' by DotNetNuke Corporation
'

Imports System
Imports DotNetNuke

Namespace DotNetNuke.Modules.Html

    Partial Public MustInherit Class DataProvider

#Region "Abstract methods"

        Public MustOverride Function AddWorkflow(ByVal PortalID As Integer, ByVal WorkflowName As String, ByVal Description As String, ByVal IsDeleted As Boolean) As Integer
        Public MustOverride Function AddWorkflowState(ByVal WorkflowID As Integer, ByVal StateName As String, ByVal Order As Integer, ByVal Notify As Boolean, ByVal IsActive As Boolean) As Integer
        Public MustOverride Function CanDeleteWorkFlowState(ByVal StateID As Integer) As Boolean
        Public MustOverride Sub DeleteWorkflowState(ByVal StateID As Integer)
        Public MustOverride Function GetWorkflow(ByVal WorkflowID As Integer) As IDataReader
        Public MustOverride Function GetWorkflowState(ByVal StateID As Integer) As IDataReader
        Public MustOverride Sub UpdateWorkflow(ByVal WorkflowID As Integer, ByVal WorkflowName As String, ByVal Description As String, ByVal IsDeleted As Boolean)
        Public MustOverride Sub UpdateWorkflowState(ByVal StateID As Integer, ByVal StateName As String, ByVal Order As Integer, ByVal Notify As Boolean, ByVal IsActive As Boolean)

        Public MustOverride Function GetPermissionsByStateID(ByVal StateID As Integer) As IDataReader
        Public MustOverride Function GetWorkflowStatePermission(ByVal WorkflowStatePermissionID As Integer) As IDataReader
        Public MustOverride Sub DeleteWorkflowStatePermissionsByStateID(ByVal StateID As Integer)
        Public MustOverride Sub DeleteWorkflowStatePermissionsByUserID(ByVal userID As Integer)
        Public MustOverride Sub DeleteWorkflowStatePermission(ByVal WorkflowStatePermissionID As Integer)
        Public MustOverride Function AddWorkflowStatePermission(ByVal StateID As Integer, ByVal permissionID As Integer, ByVal roleID As Integer, ByVal allowAccess As Boolean, ByVal userID As Integer, ByVal createdByUserID As Integer) As Integer
        Public MustOverride Sub UpdateWorkflowStatePermission(ByVal WorkflowStatePermissionID As Integer, ByVal StateID As Integer, ByVal permissionID As Integer, ByVal roleID As Integer, ByVal allowAccess As Boolean, ByVal userID As Integer, ByVal lastModifiedByUserID As Integer)

#End Region

    End Class

End Namespace

File: Trunk/Community/Modules/HTML/Components/DataProviderPro.cs
#region Copyright
// 
// DotNetNuke® - http://www.dotnetnuke.com
// Copyright (c) 2002-2012
// by DotNetNuke Corporation
// 
// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
// 
// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
// of the Software.
// 
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
// DEALINGS IN THE SOFTWARE.
#endregion
using System.Data;


namespace DotNetNuke.Modules.Html
{
    public abstract partial class DataProvider
    {
        #region Abstract methods

        public abstract int AddWorkflow(int PortalID, string WorkflowName, string Description, bool IsDeleted);

        public abstract int AddWorkflowState(int WorkflowID, string StateName, int Order, bool Notify, bool IsActive);

        public abstract bool CanDeleteWorkFlowState(int StateID);

        public abstract void DeleteWorkflowState(int StateID);

        public abstract IDataReader GetWorkflow(int WorkflowID);

        public abstract IDataReader GetWorkflowState(int StateID);

        public abstract void UpdateWorkflow(int WorkflowID, string WorkflowName, string Description, bool IsDeleted);

        public abstract void UpdateWorkflowState(int StateID, string StateName, int Order, bool Notify, bool IsActive);

        public abstract IDataReader GetPermissionsByStateID(int StateID);

        public abstract IDataReader GetWorkflowStatePermission(int WorkflowStatePermissionID);

        public abstract void DeleteWorkflowStatePermissionsByStateID(int StateID);

        public abstract void DeleteWorkflowStatePermissionsByUserID(int userID);

        public abstract void DeleteWorkflowStatePermission(int WorkflowStatePermissionID);

        public abstract int AddWorkflowStatePermission(int StateID, int permissionID, int roleID, bool allowAccess, int userID, int createdByUserID);

        public abstract void UpdateWorkflowStatePermission(int WorkflowStatePermissionID, int StateID, int permissionID, int roleID, bool allowAccess, int userID, int lastModifiedByUserID);

        #endregion
    }
}